### PR TITLE
Apns: Use upstream enum for priority

### PR DESCRIPTION
**Pull request by SMillerDev**

### CODE REVIEW

**Review by marcella-galambos-move — APPROVED** [review]

**Review by brianstoop — APPROVED** [review]

File: src/Lunr/Vortex/FCM/FCMApnsPriority.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    /**
+     * Send the notification based on power considerations on the user’s device
+     */
+    case default = 5;
+
+    /**
+     * Send the notification immediately
+     */
+    case high = 10;
```
Comment (pprkut) [inline src/Lunr/Vortex/FCM/FCMApnsPriority.php:32]: enum cases should be PascalCase

**Review by pprkut — CHANGES_REQUESTED** [review]

File: src/Lunr/Vortex/FCM/FCMApnsPriority.php (hunk 1 above)
Comment (SMillerDev) [reply src/Lunr/Vortex/FCM/FCMApnsPriority.php:32]: No stylecheck failed 😄.

I will make https://github.com/lunr-php/lunr.vortex/pull/12/files#diff-f930c4fe97c42bf5fc90693166070d3cdc690cfc79ea1b1bbc5891ace89b6f41 more complicated, but I guess that's fine.

File: src/Lunr/Vortex/FCM/FCMApnsPayload.php [hunk 2]
```diff
@@ -20,7 +17,7 @@
  *     image?: string
  * }
  * @phpstan-type FcmApnsConfig array{
- *     headers?: array<string,string>,
+ *     headers?: array<string,string|int>,
```
Comment (pprkut) [inline src/Lunr/Vortex/FCM/FCMApnsPayload.php:20]: Not according to the upstream documentation https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#ApnsConfig

`headers` is `map (key: string, value: string)`

The example has numbers also as a string there

**Review by pprkut — CHANGES_REQUESTED** [review]

**Review by pprkut — CHANGES_REQUESTED** [review]

File: src/Lunr/Vortex/FCM/FCMApnsPayload.php [hunk 3]
```diff
@@ -114,13 +111,11 @@ public function set_mutable_content(bool $mutable): static
...
-
-        $priority_class = new ReflectionClass(APNSPriority::class);
-        $priorities     = $priority_class->getConstants();
-        if (in_array($priority, array_keys($priorities)))
+        $priority = strtolower($priority);
+        $matches  = array_filter(FCMApnsPriority::cases(), fn($case) => strtolower($case->name) === $priority);
+        if (count($matches) === 1)
         {
-            $this->elements['headers']['apns-priority'] = $priorities[$priority];
+            $this->elements['headers']['apns-priority'] = (string) array_pop($matches)->value;
```
[the remark this answers is not in the document]
Comment (SMillerDev) [reply src/Lunr/Vortex/FCM/FCMApnsPayload.php:118]: So a `try_from_string`?

**Review by pprkut — APPROVED** [review]

**Pull request merged**